azure/scope/managedcontrolplane_test.go

```diff
@@ -388,7 +391,7 @@ func getAzureMachinePoolWithScaling(name string, min, max int32) *infrav1.AzureM
 }
 
 func getAzureMachinePoolWithMaxPods(name string, maxPods int32) *infrav1.AzureManagedMachinePool {
-	managedPool := getAzureMachinePool(name, infrav1.NodePoolModeUser)
+	managedPool := getAzureMachinePool(name, infrav1.NodePoolModeSystem)
```

Comment:
Why is this being updated?

Reply:
to make it a valid request. We cannot create a cluster with all machine pools with mode `User`

Reply:
isn't the exact same thing being done above for `getAzureMachinePoolWithScaling`? Should we be validating both System and User pools or are the functionally equivalent for this purpose?

Reply:
yes, we should be testing both actually. But in case of scaling its not an immutable field so we just need to make sure it works for `AgentPoolSpec()` and in this case we have to make sure it works also for `GetAgentPoolSpecs()` which is the one complaining about no system nodepool present.